### PR TITLE
Use MapVerifier in monitor

### PR DESCRIPTION
**Pull request by gdbelvin**
This PR cleans up a little bit of the monitor. Many of the more complex functions that used to be manually performed have been moved to the trillian `MapVerifier`.

### CODE REVIEW

File: core/monitor/monitor.go [hunk 1]
```diff
@@ -40,54 +37,38 @@ type Monitor struct {
...
-	logHasher, err := hashers.NewLogHasher(logTree.GetHashStrategy())
-	if err != nil {
-		return nil, fmt.Errorf("could not initialize log hasher: %v", err)
-	}
-	logPubKey, err := der.UnmarshalPublicKey(logTree.GetPublicKey().GetDer())
-	if err != nil {
-		return nil, fmt.Errorf("failed parsing log public key: %v", err)
-	}
-	mapHasher, err := hashers.NewMapHasher(mapTree.GetHashStrategy())
+
```
Comment (phad) [inline core/monitor/monitor.go]: unnecessary empty line

File: core/monitor/monitor.go [hunk 2]
```diff
@@ -40,54 +37,38 @@ type Monitor struct {
...
 	}
-	logVerifier := client.NewLogVerifier(logHasher, logPubKey)
-	return New(mclient, logVerifier,
-		mapTree.TreeId, mapHasher, mapPubKey,
-		signer, store)
+	return New(mclient, logVerifier, mapVerifier, signer, store)
 }
 
 // New creates a new instance of the monitor.
 func New(mclient pb.KeyTransparencyClient,
```
Comment (phad) [inline core/monitor/monitor.go]: nit: mClient

File: core/monitor/verify.go [hunk 3]
```diff
@@ -159,7 +152,7 @@ func (m *Monitor) verifyMutations(muts []*pb.MutationProof, oldRoot, expectedNew
 			glog.Infof("Mutation did not verify: %v", err)
 			errs.AppendStatus(status.Newf(codes.DataLoss, "invalid mutation: %v", err).WithDetails(mut.GetMutation()))
 		}
-		newLeafnID := storage.NewNodeIDFromPrefixSuffix(index, storage.Suffix{}, m.mapHasher.BitLen())
+		newLeafnID := storage.NewNodeIDFromPrefixSuffix(index, storage.Suffix{}, m.mapVerifier.Hasher.BitLen())
```
Comment (phad) [inline core/monitor/verify.go]: nit (not in your change strictly): newLeafnID is a weird name.  (Leafn ?)  leafNodeID would be more readable.
and below, just leaf in place of newLeaf.

File: core/monitor/monitor.go (hunk 1 above)
Comment (gdbelvin) [reply core/monitor/monitor.go]: done

File: core/monitor/monitor.go (hunk 2 above)
Comment (gdbelvin) [reply core/monitor/monitor.go]: done

File: core/monitor/verify.go (hunk 3 above)
Comment (gdbelvin) [reply core/monitor/verify.go]: done

**Pull request merged**